driver-core/src/test/functional/com/mongodb/operation/CreateIndexesOperationSpecification.groovy

```diff
@@ -448,6 +448,22 @@ class CreateIndexesOperationSpecification extends OperationFunctionalSpecificati
         getIndex('a_1').get('storageEngine') == storageEngineOptions
     }
 
+    @IgnoreIf({ !serverVersionAtLeast(asList(3, 1, 8)) })
+    def 'should be able to create a partially filtered index'() {
+        given:
+        def partialFilterExpression = new Document('a', new Document('$gte', 10))
+        def createIndexOperation = new CreateIndexesOperation(getNamespace(),
+                [new IndexRequest(new BsonDocument('field', new BsonInt32(1)))
+                         .partialFilterExpression(new BsonDocumentWrapper(partialFilterExpression, new DocumentCodec()))])
+
+        when:
+        createIndexOperation.execute(getBinding())
+
+        then:
+        getUserCreatedIndexes('partialFilterExpression').head() == partialFilterExpression
```

Comment:
Unfortunately this is not all that useful a test, since the server seems to do no argument checking for create indexes:

> db.runCommand({createIndexes : 'test', indexes : [{key : { y : 1}, name : 'y_1', ljsf : 1}] })
{
	"createdCollectionAutomatically" : false,
	"numIndexesBefore" : 2,
	"numIndexesAfter" : 3,
	"ok" : 1
}
> db.runCommand({listIndexes : 'test'})
{
	"cursor" : {
		"id" : NumberLong(0),
		"ns" : "test.$cmd.listIndexes.test",
		"firstBatch" : [
			{
				"v" : 1,
				"key" : {
					"_id" : 1
				},
				"name" : "_id_",
				"ns" : "test.test"
			},
			{
				"v" : 1,
				"key" : {
					"x" : 1
				},
				"name" : "x_1",
				"ns" : "test.test"
			},
			{
				"v" : 1,
				"key" : {
					"y" : 1
				},
				"name" : "y_1",
				"ns" : "test.test",
				"ljsf" : 1
			}
		]
	},

But I don't see any other way to automatically test that it actually works